azure/services/agentpools/agentpools.go

```diff
@@ -46,6 +46,7 @@ type AgentPoolScope interface {
 	SetCAPIMachinePoolReplicas(replicas *int32)
 	SetCAPIMachinePoolAnnotation(key, value string)
 	RemoveCAPIMachinePoolAnnotation(key string)
+	SetSubnetName()
```

Comment:
where is this used by the scope interface? I don't see it used below

Reply:
https://github.com/spectrocloud/cluster-api-provider-azure/blob/8f5ec54637ee3fbb6dcba201a0c1f35e8aa8c11b/controllers/azuremanagedmachinepool_reconciler.go#L98

Reply:
This is using `ManagedMachinePoolScope`, not the `AgentPoolScope` scope interface. I bet if you remove this it will still work.